### ISSUE TITLE
How to insert data when i get error: "DB::Exception: Too many parts (300). Parts cleaning are processing significantly slower than inserts."

### ISSUE THREAD

**Issue by khuenth** [ClickHouse/ClickHouse#24932]
My ClickHouse cluster's topology is: 3shards and 2 replicas, zookeeper cluster 3 nodes
My system was running perfectly until my DEV create a new table for new project. He inserted and deleted data routine in midnight. He did it without discussion. When the whole system could not insert any more with error "DB::Exception: Too many parts (300). Parts cleaning are processing significantly slower than inserts.". I knew Clickhouse could not support delete data like that. But my question is if i got that issue, how can i continue to keep inserting new data on that system? Because this system is too much data and we do not want to rebuild it. Could you support me this case? Thank you.

My Clickhouse version 21.4.7.3 (official build)

My fully error statement:
DB::Exception: Received from ch-analytic03-141-194:9000. DB::Exception: Too many parts (300). Parts cleaning are processing significantly slower than inserts. (version 21.4.7.3 (official build))

**Comment by den-crane** [comment]
You can increase 300 to 3000 and delay the issue. 
https://clickhouse.tech/docs/en/operations/settings/merge-tree-settings/#parts-to-throw-insert

**Comment by filimonov** [comment]
It's about inactive_parts_to_throw_insert

**Comment by khuenth** [comment]
> 
> 
> > Parts cleaning...
> 
> It's about inactive_parts_to_throw_insert

What will i do with that option ?

**Comment by khuenth** [comment]
I'm trying to drop all databases and create new one then import data. But the issue is still happening. Is that concern to zookeeper cluster which keep parts information ?

**Comment by den-crane** [comment]
@filimonov https://github.com/ClickHouse/ClickHouse/commit/9238d8e54aea2eb2a497d314f10f870f6f9cba5d
Amos accidentally replaced the exception message with a wrong one.

**Comment by khuenth** [comment]
For more information:
**########### This is my clickhouse configurations on each node file /etc/clickhouse-server/metrika.xml:**
Node 1:
<yandex>

    
    <clickhouse_remote_servers>
        <cluster_3s_2r>

            
            <shard>
                <weight>1</weight>
                <internal_replication>true</internal_replication>
                <replica>
                    <default_database>shard01</default_database>
                    <host>ch-analytic01</host>
                    <port>9000</port>
                    <user>default</user>
                    <password>xxx</password>
                </replica>
                <replica>
                    <default_database>shard01</default_database>
                    <host>ch-analytic02</host>
                    <port>9000</port>
                    <user>default</user>
                    <password>xxx</password>
                </replica>
            </shard>

            
            <shard>
                <weight>1</weight>
                <internal_replication>true</internal_replication>
                <replica>
                    <default_database>shard02</default_database>
                    <host>ch-analytic02</host>
                    <port>9000</port>
                    <user>default</user>
                    <password>xxx</password>
                </replica>
                <replica>
                    <default_database>shard02</default_database>
                    <host>ch-analytic03</host>
                    <port>9000</port>
                    <user>default</user>
                    <password>xxx</password>
                </replica>
            </shard>

            
            <shard>
                <weight>1</weight>
                <internal_replication>true</internal_replication>
                <replica>
                    <default_database>shard03</default_database>
                    <host>ch-analytic03</host>
                    <port>9000</port>
                    <user>default</user>
                    <password>xxx</password>
                </replica>
                <replica>
                    <default_database>shard03</default_database>
                    <host>ch-analytic01</host>
                    <port>9000</port>
                    <user>default</user>
                    <password>xxx</password>
                </replica>
            </shard>
        </cluster_3s_2r>
    </clickhouse_remote_servers>

    
    <zookeeper-servers>
        <node index="1">
            <host>zk-analytic01</host>
            <port>2181</port>
        </node>
        <node index="2">
            <host>zk-analytic02</host>
            <port>2181</port>
        </node>
        <node index="3">
            <host>zk-analytic03</host>
            <port>2181</port>
        </node>
    </zookeeper-servers>
<macros>
     <layer>01</layer>
     <shard>01</shard>
     <replica>repl01</replica>
     <layer>01</layer>
     <shard>03</shard>
     <replica>repl02</replica>
</macros>
    <networks>
        <ip>::/0</ip>
    </networks>

    
    <clickhouse_compression>
        <case>
            <min_part_size>10000000000</min_part_size>
            <min_part_size_ratio>0.01</min_part_size_ratio>
            <method>lz4</method>
        </case>
    </clickhouse_compression>

</yandex>

**############## Tag in each nodes are same, online tag <marcros> are different:**
------- Node1:
<macros>
     <layer>01</layer>
     <shard>01</shard>
     <replica>repl01</replica>
     <layer>01</layer>
     <shard>03</shard>
     <replica>repl02</replica>
</macros>

------- Node 2:
<macros>
     <layer>01</layer>
     <shard>02</shard>
     <replica>repl01</replica>
     <layer>01</layer>
     <shard>01</shard>
     <replica>repl02</replica>
</macros>

------- Node 3:
<macros>
     <layer>01</layer>
     <shard>03</shard>
     <replica>repl01</replica>
     <layer>01</layer>
     <shard>02</shard>
     <replica>repl02</replica>
</macros>

**#################### This is my /etc/clickhouse-server/users.xml**

<?xml version="1.0"?>
<yandex>
    
    <profiles>
        
        <default>
            
            <max_memory_usage>10000000000</max_memory_usage>
        <max_partitions_per_insert_block>1000</max_partitions_per_insert_block>
        <distributed_product_mode>allow</distributed_product_mode>
        <insert_distributed_sync>1</insert_distributed_sync>
            
            <load_balancing>random</load_balancing>
        </default>
        <allow_experimental_map_type>1</allow_experimental_map_type>

        
        <readonly>
            <readonly>1</readonly>
        </readonly>
    </profiles>

    
    <users>
        
        <default>
            
            <password></password>

            
            <networks>
                <ip>::/0</ip>
            </networks>

            
            <profile>default</profile>

            
            <quota>default</quota>

            
            
        </default>
    </users>

    
    <quotas>
        
        <default>
            
            <interval>
                
                <duration>3600</duration>

                
                <queries>0</queries>
                <errors>0</errors>
                <result_rows>0</result_rows>
                <read_rows>0</read_rows>
                <execution_time>0</execution_time>
            </interval>
        </default>
    </quotas>
</yandex>

**Comment by khuenth** [comment]
I found out what problem i got. I tried to query the table system.replication_queue. It showed many row with last_exception col which show my issue. My issue is : **Code: 44, e.displayText() = DB::Exception: There is no subcolumn format in type Nested(topic Int32) (version 21.4.7.3 (official build))**
I tried to change the structure of table. It fixed. Merge parts smoothly.
Thank you for your support, Crane and Filimonov!
